src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_safe_repr.py

```diff
@@ -279,8 +281,54 @@ def _repr_dict(self, obj, level, prefix, suffix,
         yield suffix
 
     def _repr_str(self, obj, level):
-        return self._repr_obj(obj, level,
-                              self.maxstring_inner, self.maxstring_outer)
+        try:
+            if self.raw_value:
+                # For raw value retrieval, ignore all limits.
+                if isinstance(obj, bytes):
+                    yield obj.decode('latin-1')
+                else:
+                    yield self._convert_to_unicode_or_bytes_repr(repr(obj))
+                return
+
+            limit_inner = self.maxother_inner
+            limit_outer = self.maxother_outer
+            limit = limit_inner if level > 0 else limit_outer
+            if len(obj) <= limit:
+                # Note that we check the limit before doing the repr, so, this could account
+                # for up to 3 chars above the limit afterwards. i.e.: u'' or b''
+                yield self._convert_to_unicode_or_bytes_repr(repr(obj))
+                return
+
+            # Slightly imprecise calculations - we may end up with a string that is
+            # up to 6 characters longer than limit. If you need precise formatting,
+            # you are using the wrong class.
+            left_count, right_count = max(1, int(2 * limit / 3)), max(1, int(limit / 3))  # noqa
+
+            # Important: only do repr after slicing to avoid duplicating a byte array that could be
+            # huge.
+
+            # Note: we don't deal with high surrogates here because we're not dealing with the
+            # repr() of a random object (which we have to convert to unicode to show as unicode
+            # when needed), we just show the unicode/string/bytes repr() directly to make clear
```

Comment:
On Python 2, can't Unicode strings have surrogate pairs in them? So then the slices can potentially end up splitting the pair...

Reply:
It may split the pair, but as we do a `repr` afterwards, that's ok (i.e.: it's different from doing a `repr` on a random object where the returned string won't have an additional `repr` done it in -- in which case we do have to deal with high surrogates).